microsoft/testsuites/power/power.py

```diff
@@ -0,0 +1,680 @@
+        _ret = node.tools[Cat].read_with_filter(
+            "/var/log/messages",
+            message,
+            sudo=True
+        )
+    elif node.shell.exists('/var/log/syslog'):
+        _ret = node.tools[Cat].read_with_filter(
+            "/var/log/syslog",
+            message,
+            sudo=True
+        )
+    else:
+        _ret = node.execute(
+            f'dmesg | grep -i {message}',
+            shell=True,
+            sudo=True
+        ).stdout
+    if message in _ret:
+        return True
+    else:
+        return False
+
+
+def _is_vm_running(node: Node, log: Logger, timeout: int = 300) -> bool:
+    is_ready, tcp_error_code = wait_tcp_port_ready(
+        node.public_address,
+        node.public_port,
+        log=log,
+        timeout=timeout,
+    )
+    return is_ready
+
+
+def _recover_files(node: Node, log: Logger, files_and_contents: dict) -> None:
+    for file in files_and_contents:
+        node.tools[Echo].write_to_file(
+            files_and_contents[file],
+            file,
+            sudo=True
+        )
+        log.debug(f'Recovering file: {file} with content: {files_and_contents[file]}')
+
+
+@TestSuiteMetadata(
+    area="power",
+    category="functional",
+    description="""
+    """
+)
+class Power(TestSuite):
+    REBOOT_TIME_OUT = 300
+
+    @TestCaseMetadata(
+        description="""
+        This test can be performed in Azure and Hyper-V both.
+        But this script only covers Azure.
+        1. Prepare swap space for hibernation
+        2. Update the grub.cfg with resume=UUID=xxxx where is from blkid swap disk
+        3. Set RTC to future time or past time
+        4. Hibernate the VM
+        5. Resume the VM
+        6. Read RTC timestamp and compare to the original value
+        """,
+        priority=4,
+        requirement=simple_requirement(
+            disk=schema.DiskOptionSettings(
+                disk_type=schema.DiskType.StandardSSDLRS,
+                data_disk_iops=search_space.IntRange(min=2000),
+                data_disk_count=search_space.IntRange(min=1),
+            ),
+        )
+    )
+    def power_clock_sync_hibernate(
+        self,
+        environment: Environment,
+        node: Node,
+        log: Logger
+    ) -> None:
+        modified_files = {}
+
+        dmesg = node.tools[Dmesg]
+        dmesg_output = dmesg.get_output(force_run=True)
+        search_pattern = re.compile(r'root=.*resume', flags=re.M | re.IGNORECASE)
+        env_setup_result = re.findall(search_pattern, dmesg_output)
+        log.debug("Env Setup Result: " + ''.join(env_setup_result))
+        if env_setup_result:
+            log.info("Evironment has been set up...")
+        else:
+            # Configuration for hibernation
+            node.execute(
+                "umask 022"
+            )
+
+            # GetGuestGeneration
+            os_generation = node.tools[VmGeneration].get_generation()
+
+            if isinstance(node.os, Redhat):
```

Comment:
we can skip the case in the very beginning if current distro/kernel doesn't support hibernation at all.

Reply:
moved the check to the beginning of the test